src/mscorlib/src/System/Runtime/Intrinsics/X86/Avx.cs

```diff
@@ -238,42 +239,145 @@ public static class Avx
         /// __int8 _mm256_extract_epi8 (__m256i a, const int index)
         ///   HELPER
         /// </summary>
-        public static sbyte Extract(Vector256<sbyte> value, byte index) => Extract(value, index);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static sbyte Extract(Vector256<sbyte> value, byte index)
+        {
+            index &= 0x1F; // the instructions only nedd the lowest 5 bits.
```

Comment:
I think we still need some more comments in these functions.

It takes some parsing to read already, and if someone isn't familiar with intrinsics, they might not have any clue what is going on.

Explicitly calling out that branch here covers upper vs lower bits would be useful, for example. This also makes it more clear why you are subtracting 16 from the index (because you are accessing the upper bits as if they started at index 0).

Reply:
Let me improve the comments in the next PR, maybe I will remove the managed-implementation and move it into JIT for better CQ.